include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -182,33 +182,73 @@ class concatenated_sequences
      * \brief Static constexpr variables that emulate/encapsulate seqan3::compatible (which doesn't work for types during their definition).
      * \{
      */
+    //!\cond
+    // unfortunately we cannot specialise the variable template so we have to add an auxiliary here
+    template <std::ranges::range t>
+        requires std::convertible_to<std::ranges::range_reference_t<t>, std::ranges::range_value_t<value_type>>
+    static constexpr bool is_compatible_with_value_type_aux = dimension_v<t> == dimension_v<value_type>;
+    //!\endcond
+
     //!\brief Whether a type satisfies seqan3::compatible with this class's `value_type` or `reference` type.
     //!\hideinitializer
     // we explicitly check same-ness, because these types may not be fully resolved, yet
-    template <typename t>
-    static constexpr bool is_compatible_value = std::is_same_v<remove_cvref_t<t>, value_type>       ||
-                                                std::is_same_v<remove_cvref_t<t>, reference>        ||
-                                                std::is_same_v<remove_cvref_t<t>, const_reference>  ||
-                                                (dimension_v<t> == dimension_v<value_type> &&
-                                                std::convertible_to<reference_t<t>, value_type_t<value_type>>);
-    //!\}
+    template <std::ranges::range t>
+    static constexpr bool is_compatible_with_value_type =
+        std::is_same_v<remove_cvref_t<t>, value_type> ||
+        std::is_same_v<remove_cvref_t<t>, reference> ||
+        std::is_same_v<remove_cvref_t<t>, const_reference> ||
+        (
+            !std::same_as<remove_cvref_t<t>, iterator> &&
+            !std::same_as<remove_cvref_t<t>, const_iterator> &&
+            !std::same_as<remove_cvref_t<t>, concatenated_sequences> &&
+            is_compatible_with_value_type_aux<t>
+        );
 
     //!\cond
     // unfortunately we cannot specialise the variable template so we have to add an auxiliary here
     template <typename t>
-        requires (dimension_v<t> == dimension_v<value_type> + 1) &&
-                  is_compatible_value<reference_t<t>>
-    static constexpr bool is_compatible_this_aux = true;
+        requires std::ranges::range<std::iter_reference_t<t>> &&
+                 (dimension_v<t> == dimension_v<value_type> + 1) &&
```

Comment:
`t` is an iterator and can have a dimension?

Reply:
yes, right now the dimension is defined as:

https://github.com/seqan/seqan3/blob/35f3efb613c9cb039f227f38aba207aa49ecceb4/include/seqan3/core/type_traits/range.hpp#L196-L216

This behaviour is the same as before and yeah, I know, that this is utterly confusing.

This has to be fixed somehow. 

Maybe by `dimension_v<t> == dimension_v<std::iter_reference_t<t>> + 1 == dimension_v<value_type> + 1`

Reply:
>  I know, that this is utterly confusing.

If just that. :) I think it was a really good decion fromt the standard to seperate the type traits between iterators and ranges and we should adhere to that. But ok, it was like so before, so we can keep it.